File: Claude/1.Algoritmo Básico Ordenamiento por Burbuja/ruby/burbuja.rb
# Clase para implementar ordenamiento burbuja
class BubbleSort
  # Implementación del algoritmo de ordenamiento burbuja
  # @param arr [Array<Integer>] Array de enteros a ordenar
  # @return [Array<Integer>] Array ordenado
  def self.sort(arr)
    sorted_arr = arr.dup
    n = sorted_arr.length
    
    (0...n-1).each do |i|
      swapped = false
      (0...n-i-1).each do |j|
        if sorted_arr[j] > sorted_arr[j+1]
          sorted_arr[j], sorted_arr[j+1] = sorted_arr[j+1], sorted_arr[j]
          swapped = true
        end
      end
      break unless swapped
    end
    
    sorted_arr
  end
end

# Ejemplo de uso
numbers = [64, 34, 25, 12, 22, 11, 90]
sorted_numbers = BubbleSort.sort(numbers)

puts "Array original: #{numbers}"
puts "Array ordenado: #{sorted_numbers}"

File: Claude/1.Algoritmo Básico Ordenamiento por Burbuja/javascript/burbuja.js
/**
 * Implementación del algoritmo de ordenamiento burbuja
 * @param {number[]} arr - Array de números enteros
 * @returns {number[]} Array ordenado
 */
function bubbleSort(arr) {
    const sortedArr = [...arr];
    const n = sortedArr.length;
    
    for (let i = 0; i < n - 1; i++) {
        let swapped = false;
        for (let j = 0; j < n - i - 1; j++) {
            if (sortedArr[j] > sortedArr[j + 1]) {
                [sortedArr[j], sortedArr[j + 1]] = [sortedArr[j + 1], sortedArr[j]];
                swapped = true;
            }
        }
        if (!swapped) break;
    }
    return sortedArr;
}

// Ejemplo de uso
const numbers = [64, 34, 25, 12, 22, 11, 90];
const sorted = bubbleSort(numbers);
console.log('Array original:', numbers);
console.log('Array ordenado:', sorted);